### DISCUSSION TITLE
【依頼】TOPPERSマスター方法について

### DISCUSSION THREAD

**Discussion by tsukada-shigeki** [toppers/users-forum#79] [TOPPERS Seminar]
竹内様

お世話になります。
１）TOPPERS基礎を楽しく受講できました。有難うございます。
２）RTOSは興味が有りますが、実務レベルまでレベルを上げるために今後どのようなアプローチをすればよろしいでしょうか？良い方法があればご紹介願います。
TOPPERS会員になればブラッシュアップ可能でしょうか？
３）下記実績を拝見しましたが、TOPPERSは組み込みソフト開発の主流になっているのでしょうか？
https://www.toppers.jp/docs/toppers-muranaka-0505.pdf
お忙しいところ誠に申し訳ありませんが、ご意見をお願い致します。
塚田

**Comment by Ryohsuke-Takeuchi** [comment]
個人的な意見による回答となります。
組込みソフトウェアの技術力を上げるには、
１．ハードウェアの知識と経験をもつこと
２．手法や技法ではなく、自分自身の力、または、優秀なソフト技術者の意見をよく聞き、開発経験を積むこと
３．アルゴリズムを理解し、上質なプログラムを高速に作成できる能力

３については、万国共通、どの分野のソフトウェア技術者でも重要な能力だと思います。
開発手法やコーディングルールや方法論は、多人数で組込み開発を行う場合に必要となる技術で
３の能力が高い技術者ならば、自然に、最適化されたプログラムは作れると思います。
TOPPERS会員については、いろいろな分野の技術者とかかわって、自分自身をエスカレーションできるという点では良いと思います
TOPPERSに関して（あくまでも私的な意見ですが）、RTOSは組込みプラットフォームを作るためタスク化のためのパーツです。
組込みプラットフォームを作るタスク化の部品としてTOPPERSを見た場合、FreeRTOSの方が考え方やプラットフォーム化については、優れていると思います。ガラパゴス化ですかね。

**Reply by tsukada-shigeki** [reply]
竹内様
お世話になります。
お忙しい中貴重なご意見有難うございました。
FreeRTOSに関して知識が有りませんので調べます。
塚田
